secrets-plugin/src/main/java/io/airlift/secrets/keystore/KeystoreSecretProvider.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.secrets.keystore;
+
+import com.google.inject.Inject;
+import io.airlift.spi.secrets.SecretProvider;
+import io.airlift.spi.secrets.SecretProviderException;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.KeyStore;
+
+import static io.airlift.secrets.keystore.KeystoreUtils.loadKeyStore;
+import static io.airlift.secrets.keystore.KeystoreUtils.readEntity;
+
+public class KeystoreSecretProvider
+        implements SecretProvider
+{
+    private final KeyStore keyStore;
+    private final String keystorePassword;
+
+    @Inject
+    public KeystoreSecretProvider(KeystoreSecretProviderConfig config)
+            throws GeneralSecurityException, IOException
+    {
+        keyStore = loadKeyStore(config.getKeyStoreType(), config.getKeyStoreFilePath(), config.getKeyStorePassword());
+        this.keystorePassword = config.getKeyStorePassword();
+    }
+
+    @Override
+    public String generateSecretValue(String key)
+    {
+        try {
+            return readEntity(keyStore, key, keystorePassword);
+        }
+        catch (GeneralSecurityException e) {
+            throw new SecretProviderException(e.getMessage(), e);
```

Comment:
No need to pass both the message and the cause. Passing the cause will fill in the message.